opt_einsum/contract.py

```diff
@@ -15,6 +15,68 @@
 __all__ = ["contract_path", "contract", "format_const_einsum_str", "ContractExpression", "shape_only", "shape_only"]
 
 
+class PathInfo:
+
+    def __init__(self, contraction_list, input_subscripts, output_subscript,
+                 indices, scale_list, naive_cost, opt_cost, size_list):
+        self.contraction_list = contraction_list
+        self.input_subscripts = input_subscripts
+        self.output_subscript = output_subscript
+        self.indices = indices
+        self.scale_list = scale_list
+        self.naive_cost = naive_cost
+        self.opt_cost = opt_cost
+        self.largest_intermediate = max(size_list)
+
+    def __repr__(self):
+        from decimal import Decimal
+
+        # naive costs / speedups can easily reach >~ 1e308, need exact arithmetic
+        naive_cost = Decimal(self.naive_cost)
+        opt_cost = Decimal(self.opt_cost)
+        speedup = naive_cost / opt_cost
+        largest_intermediate = Decimal(self.largest_intermediate)
+
+        # Return the path along with a nice string representation
+        overall_contraction = self.input_subscripts + "->" + self.output_subscript
+        header = ("scaling", "BLAS", "current", "remaining")
+
+        path_print = "  Complete contraction:  {}\n".format(overall_contraction)
+        path_print += "         Naive scaling:  {}\n".format(len(self.indices))
+        path_print += "     Optimized scaling:  {}\n".format(max(self.scale_list))
+        path_print += "      Naive FLOP count:  {:.3e}\n".format(naive_cost)
+        path_print += "  Optimized FLOP count:  {:.3e}\n".format(opt_cost)
+        path_print += "   Theoretical speedup:  {:3.3f}\n".format(speedup)
+        path_print += "  Largest intermediate:  {:.3e} elements\n".format(largest_intermediate)
+        path_print += "-" * 80 + "\n"
+        path_print += "{:>6} {:>11} {:>22} {:>37}\n".format(*header)
+        path_print += "-" * 80
+
+        for n, contraction in enumerate(self.contraction_list):
+            inds, idx_rm, einsum_str, remaining, do_blas = contraction
+            remaining_str = ",".join(remaining) + "->" + self.output_subscript
+            path_run = (self.scale_list[n], do_blas, einsum_str, remaining_str)
+            path_print += "\n{:>4} {:>14} {:>22} {:>37}".format(*path_run)
+
+        return path_print
```

Comment:
nit: you could avoid quadratic growth by using `lines.append(...)` and finally `return '\n'.join(lines)`